test/functionalTest/testHarness.sh

```diff
@@ -731,22 +764,54 @@ do
     continue
   fi
 
-  if [ "$verbose" == "off" ]
-  then
-    init=$testFile" ................................................................................................................."
-    init=${init:0:110}
-    printf "%03d/%d: %s " "$testNo" "$noOfTests" "$init"
-  else
-    printf "Running test %03d/%d: %s\n" "$testNo" "$noOfTests" "$testFile"
-  fi
-
   testNo=$testNo+1
   startDate=$(date)
   start=$(date --date="$startDate" +%s)
   endDate=""
+  typeset -i tryNo
+  tryNo=1
+
   if [ "$dryrun" == "off" ]
   then
-    runTest $testFile
+    while [ $tryNo -le $MAX_TRIES ]
+    do
+      if [ "$verbose" == "off" ]
+      then
+        tryNoInfo=""
+        if [ $tryNo != "1" ]
+        then
+           tryNoInfo="(intent $tryNo)"
+        fi
+
+        init=$testFile" ................................................................................................................."
+        init=${init:0:110}
+        printf "%03d/%d: %s %s " "$testNo" "$noOfTests" "$init" "$tryNoInfo"
+      else
+        printf "Running test %03d/%d: %s\n" "$testNo" "$noOfTests" "$testFile"
+      fi
+
+      runTest $testFile $tryNo
+      if [ $shellResult == "0" ]
+      then
+        if [ $tryNo != 1 ]
+        then
+          echo "OK"
```

Comment:
Is the trial number printed in the case of 2nd and 3rd trial as part as the output line corresponding to the test? Maybe it can be interred from the logic script, but it is a bit hard to follow to me...